src/sfctl/params.py

```diff
@@ -259,3 +259,6 @@ def custom_arguments(self, _):  # pylint: disable=too-many-statements
         arg_context.argument('gateway_resource_name',
                              options_list=('-n', '--name'),
                              help="The name of the gateway resource")
+    with ArgumentsContext(self, 'mesh deployment create') as arg_context:
```

Comment:
Can you add a comment inline for what this is supposed to do? Is this particular path being hit during testing? argcomplete is a new dependency, and we have not added it to our dependency list it looks like. Thanks!

Reply:
We wanted to have consistent behavior in how parameters are accepted in both the cli s. 
Currently az cli uses this argument context for accepting the parameters. So we are using the same.

Argcomplete is already a dependency in knack: https://github.com/Microsoft/knack/blob/master/setup.py#L15 . So I don't think we need to explicitly add it again..